bip-0054/test_vectors/README.md

```diff
@@ -0,0 +1,102 @@
+
+The test vector file features a JSON array of JSON objects, each corresponding to a test case. Each
+JSON object features the following entries:
+- `header_chain`: a JSON array of strings. An ordered list of hex-encoded mainnet block headers.
+- `valid`: a JSON boolean. Whether this chain of headers is valid according to BIP54.
+- `comment`: a JSON string. Description of the test case.
+
+For the purpose of testing a Timewarp fix, a Timewarp attack was included early on in the history of
+testnet3. An implementer of BIP54 may want to ensure that syncing testnet3 by enforcing BIP54 since
+genesis will treat block `00000000118da1e2165a19307b86f87eba814845e8a0f99734dce279ca3fb029` as
+invalid.
+
+
+### Long block validation time
+
+The [`sigops.json`](sigops.json) file contains test vectors for the limit on the number of
+potentially-executed legacy signature operations in a single transaction, introduced by BIP54 in
+order to mitigate long block validation times. Each test case represents a transaction and whether a
+block containing it would be valid according to BIP54. The test cases feature an extensive set of
+combinations of inputs and output types, ways to run into the limit, historical violations and some
+pathological transactions exhibiting the specific implementation details. All test cases but those
+belonging to this last category feature transactions that are valid under current Bitcoin consensus
+rules. Each test case also features a comment describing why the transaction is (in)valid according
+to BIP54.
+
+The test vector file features a JSON array of JSON objects, each corresponding to a test case. Each
+JSON object features the following entries:
+- `spent_outputs`: a JSON array of strings. An ordered list of hex-encoded Bitcoin-serialized
+  transaction outputs spent by each input of this test case's transaction.
+- `tx`: a JSON string. A hex-encoded Bitcoin-serialized transaction to be evaluated.
+- `valid`: a JSON boolean. Whether this transaction is valid according to current consensus rules
+  supplemented by BIP54.
+- `comment`: a JSON string. Description of the test case.
+
+
+### Merkle tree malleability with 64-byte transactions
+
+The [`txsize.json`](./txsize.json) file contains test cases exercising the new constraint on
+non-witness transaction size introduced in BIP54. Each test case contains a transaction and whether
+it would be valid according to BIP54, as well as a comment describing why it is (in)valid. All test
+cases are otherwise valid according to current Bitcoin consensus rules.
+
+The test vector file features a JSON array of JSON objects, each corresponding to a test case. Each
+JSON object features the following entries:
+- `tx`: a JSON string. A hex-encoded Bitcoin-serialized transaction to be evaluated.
+- `valid`: a JSON boolean. Whether this transaction is valid according to BIP54.
+- `comment`: a JSON string. Description of the test case.
```

Comment:
I had a read of the `sigops.json` but I cannot comment directly on it because it is considered as a “diff too large” by Github.

On the test vector: "Mixed input types reaching exactly 2500 BIP54-sigops + a future Segwit program input”, I think it could be valuable to multiply it for the whole range of future Segwit program (i.e from OP_02 to OP_16) to ensure the `CheckSigopsBIP54` parser works well.